### PR TITLE
Solidity style guide improvement

### PR DESCRIPTION
**Pull request by pgev**
Solidity style guide improvement: documentation sections, constants, alignment of multiline documentation within @param and @return tags.

### CODE REVIEW

File: SOLIDITY_STYLE_GUIDE.md [hunk 1]
```diff
@@ -500,14 +501,21 @@ Hence, functions should be grouped according to their visibility and ordered:
 - internal functions
 - private functions
 
+The suggested order for variables' declarations is:
+
+- public variables
+- internal variables
+- internal variables
```
Comment (schemar) [inline SOLIDITY_STYLE_GUIDE.md]: `private`?

File: SOLIDITY_STYLE_GUIDE.md [hunk 2]
```diff
@@ -500,14 +501,21 @@ Hence, functions should be grouped according to their visibility and ordered:
...
 
 To read contract top down the following order is suggested:
 
-- using
-- event
-- enum
-- struct
+- usings
+- events
+- constants
```
Comment (schemar) [inline SOLIDITY_STYLE_GUIDE.md]: Is there a reason why the constants come after the events? I am used to having constants at the top of everything. Though I don't know why. This is also good, just wanted to know if there is a specific reason.

File: SOLIDITY_STYLE_GUIDE.md [hunk 3]
```diff
@@ -1630,6 +1638,47 @@ function commitStateRoot(uint256 _blockHeight, bytes32 _stateRoot)
 
 ### Alignment
 
+Align @param and @return tag documentation including parameter name and named
```
Comment (schemar) [inline SOLIDITY_STYLE_GUIDE.md]: Should `@param` and `@return` have backticks around them?

File: SOLIDITY_STYLE_GUIDE.md (hunk 2 above)
Comment (pgev) [reply SOLIDITY_STYLE_GUIDE.md]: Events naturally seems to be an external interface.
Constants could also include private, internal. That's why put it as high as possible in order, only after events and usings :).

**Pull request merged**